text/0015-status-update.md

```diff
@@ -0,0 +1,397 @@
+  general approach that does more transient allocation for subtables etc, and
+  then a subsetting approach that (for instance) uses a long-running process to
+  handle multiple subsetting requests, and reuses memory and other state between
+  them.
+
+## next steps
+
+### a design for mutability
+
+My main concern right now is figuring out a model for compilation & mutability.
+For compilation, I'm currently imagining generating builder types, but I haven't
+dug into this yet.
+
+Offset resolution and rewriting will also be an important consideration. A
+compiler author should be able to use these tools to do things like deduplicate
+identical subtables, or to reorder tables as needed, which means we do need a
+mechanism for rewriting offsets as needed. There are various options here; for
+instance we could track offsets by ids, and then keep track of the location of
+the offsets as we write tables into a shared buffer, rewriting them later as
+needed. This will need more thought, experimentation, and discussion.
+
+### evaluating current performance
+
+It may also be useful to get some concrete numbers on the efficiency of the
+current work-to-date. I have spent a bit of time looking at disassembly and am
+generally confident that we should be able to achieve good results, but I
+haven't *demonstrated* this.
+
+I'm not sure this is strictly necessary at this point, but it might be helpful;
+if performance is actually terrible for some unforeseen reason, we might decide
+that our approach needs to be fundamentally rethought.
+
+I've also resisted this because I'm not sure of exactly what we should measure.
+In any case, it is something we will want to do at some point.
+
+### further refinements
+
+Finally, there are various additional features and functionality that it might
+be worth exploring, such as improving type information related to offsets (for
+instance encoding in the type of the offset what sort of table it should
+produce, or what table it is indexed into.) In addition, I am *very curious*
+about looking into migrating from doing codegen in a proc-macro to doing codegen
+as a separate build script that outputs normal rust files that are then included
+and used like any other. This has several advantages: it should reduce
+compilation time, but it should also produce more readable code: currently we
+generate lots of methods and type definitions that do not actually exist in any
+of our source files. This makes it harder for users to understand what is going
+on, and it also confuses tools like IDEs, which often cannot provide
+autocomplete for these generated methods.
+
+## To discuss
+
+- Is it worth doing some real-world profiling now? For instance, we could pretty
+  easily implement something like "iterate all glyphs and manually compute
+  bounding boxes", and we could try and compare that against HarfBuzz as well as
+  other Rust font parsing implementations.
```

Comment:
HarfBuzz is **the** target for shaping. I wouldn't stress other Rust implementations all that much because afaik none of them ship at scale. Potentially interesting to help understand if we're better/worse certainly, but at the end of the day shaping vs HarfBuzz is the critical comparison.